lib/timers.js

```diff
@@ -204,8 +233,27 @@ function listOnTimeout() {
       domain.enter();
     }
 
+    // TODO(trevnorris): The check for asyncId shouldn't be necessary, but
+    // because Socket has a custom implementation of _onTimeout() and this is
+    // not yet implemented correctly we'll skip those for now.
+    // TODO(trevnorris): There is a performance hit of always needing to place
+    // the current TimerWrap id on the hooks stack. We know that TimerWrap is
+    // only a mechanism to fire these callbacks, so instead manually set the
+    // correct current id field from here.
```

Comment:
Is this block of comments now outdated?

net sockets still go through `insert()` via `_unrefActive()`.